### DISCUSSION TITLE
Multiple volume groups with the same device class on the same node

### DISCUSSION THREAD

**Discussion by PeterJanRoes** [topolvm/topolvm#750] [Q&A]
We have different nodes with different types and amounts of disks. We deploy LVMD as a DaemonSet. After searching quite a bit I discovered the `lvmd.additionalConfigs` option (I think it is not present in the documentation) to be able to configure nodes differently and use different volume groups for each node. Some nodes have both HDD and SSD disks, others have two HDD disks. I want workloads to be able to select either `hdd` or `ssd` storage by specifying the correct storage class. For nodes with two HDD disks I am trying to add both disks with the same device class name `hdd`. However, it seems that TopoLVM does not allow this. I can see two options around it but both do not seem ideal:

* I could create an extra device class with a different name but I suspect that workloads also have to specifically reference that device class (through a distinct storage class). This defeats the purpose of hiding this detail from the workload.
* I could try to combine both disks in LVM into a single volume group. The downside to this seems that if one disk fails, the whole group fails, and data may be spread across disks.
 
What would be the correct way to add two HDD disks (volume groups) on the same node so they can be used in the same storage class? Or is this not possible (yet) in TopoLVM? Am I missing a crucial insight here?

**Comment by pluser** [comment]
@PeterJanRoes 
Hello.
As you may already know from your research, TopoLVM does not include the feature you have inquired about. Storage class and volume group are tied one-to-one in TopoLVM.

**Comment by venganesh** [comment]
@pluser I have the same use case as @PeterJanRoes mentioned above. Wondering, is there a quick way to achieve this without or with minimal code changes?

**Reply by pluser** [reply]
@venganesh We don't plan to implement such a feature because it's an edge case. In addition, its implementation cost will be quite high and will complicate the code very much.

**Comment by briend** [comment]
We have a similar requirement to keep disk IO from a particular pod (from statefulsets) confined to a single disk (for QoS and fault isolation).  I can appreciate the complexity creeping into topolvm, so we'll have to continue using the [local-static-volume-provisioner](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner) and pre-create PVs of particular sizes. This lets us map multiple PVs on a single node to the same StorageClass, at the cost of being less flexible with allocating right-sized volumes, and missing other nice features of topolvm.
